src/core/thread/mle_router.cpp

```diff
@@ -2153,6 +2163,25 @@ otError MleRouter::UpdateChildAddresses(const AddressRegistrationTlv &aTlv, Chil
         }
     }
 
+    // Put back MlEid if no MlEid entry in AddressRegistration TLV
+    if (!withMlEid && !mleid.IsUnspecified())
```

Comment:
Putting MlEid back in the end would work fine if the parent has enough capacity to cache all addresses one child may register. I am not quite sure, but should we prioritize putting MlEid back than other addresses as actually the parent has limited resources for each child? @jwhui

Reply:
Interesting! Check https://github.com/openthread/openthread/pull/2504.

Reply:
Yes, #2504 should simplify this.